implementations/elixir/ockam/ockam/lib/ockam/node.ex

```diff
@@ -1,86 +1,124 @@
-    Registry.start_link(keys: :unique, name: __MODULE__)
-  end
+  @ping <<0>>
+  @pong <<1>>
 
   @doc """
-  Registers the address of a `pid`.
+  Returns the process registry for this node.
   """
-  def register(address, pid), do: register_name(address, pid)
-
-  # This function is used when a process is registed using the `:via` option.
-  #
-  # The Gen* modules expect this function to be exported.
-  # See the "Name registration" section of the `GenServer` module.
-  @doc false
-  def register_name(address, pid), do: Registry.register_name({__MODULE__, address}, pid)
+  def process_registry, do: Registry
 
   @doc """
-  Returns the `pid` of registered address, or `nil` is
+  Returns the `pid` of registered address, or `nil`.
   """
   def whereis(address) do
-    case whereis_name(address) do
+    case Registry.whereis_name(address) do
       :undefined -> nil
       pid -> pid
     end
   end
 
-  # This function is used when a process is registed using the `:via` option.
-  #
-  # The Gen* modules expect this function to be exported.
-  # See the "Name registration" section of the `GenServer` module.
-  @doc false
-  def whereis_name(address), do: Registry.whereis_name({__MODULE__, address})
+  @doc """
+  Registers the address of a `pid`.
+  """
+  defdelegate register_address(address, pid), to: Registry, as: :register_name
 
   @doc """
   Unregisters an address.
   """
-  def unregister(address), do: unregister_name(address)
-
-  # This function is used when a process is registed using the `:via` option.
-  #
-  # The Gen* modules expect this function to be exported.
-  # See the "Name registration" section of the `GenServer` module.
-  @doc false
-  def unregister_name(address), do: Registry.unregister_name({__MODULE__, address})
+  defdelegate unregister_address(address), to: Registry, as: :unregister_name
 
-  # This function is used when a process is registed using the `:via` option.
-  #
-  # The Gen* modules expect this function to be exported.
-  # See the "Name registration" section of the `GenServer` module.
-  @doc false
-  def send(address, message), do: Registry.send({__MODULE__, address}, message)
+  @doc """
+  Send a message to the process registered with an address.
+  """
+  defdelegate send(address, message), to: Registry
 
   @doc """
-  Returns a random address that is currently not registed with the node.
+  Returns a random address that is currently not registed on the node.
   """
-  def get_unused_address(length_in_bytes \\ @default_address_length_in_bytes) do
+  def get_random_unregistered_address(length_in_bytes \\ @default_address_length_in_bytes) do
     cadidate = :crypto.strong_rand_bytes(length_in_bytes)
 
     case whereis(cadidate) do
       nil -> cadidate
-      _pid -> get_unused_address(length_in_bytes)
+      _pid -> get_random_unregistered_address(length_in_bytes)
+    end
+  end
+
+  @doc false
+  def start_supervised(module, options) do
```

Comment:
would we benefit from Telemetry added to this method (and `start_link` and `init`) ? maybe it is done behind the scenes.

Reply:
oh, I see some if it is done later.

Reply:
the idea is to have telemetry be an optional feature/dependency in the `ockam` lib .. an application like `ockam_hub` will start it as part of its supervision tree.